### PR TITLE
Add DisMaxQueryFactory

### PR DESCRIPTION
**Pull request by nelsonfrz**
This PR adds DisMaxQueryFactory which is designed to build Elasticsearch `DisMax` queries, supporting both master and variant field.

### CODE REVIEW

File: search-service/src/main/java/de/cxp/ocs/elasticsearch/query/builder/DisMaxQueryFactory.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    private final Map<QueryBuildingSetting, String> queryBuildingSettings = new HashMap<>();
+    private final Map<String, Float> masterFields = new HashMap<>();
+    private final Map<String, Float> variantFields = new HashMap<>();
+    @Getter
+    @Setter
+    private String name;
+
+    @Override
+    public void initialize(String name, Map<QueryBuildingSetting, String> settings, Map<String, Float> fieldWeights, FieldConfigAccess fieldConfig) {
+        if (name != null) this.name = name;
```
Comment (gabauer) [inline search-service/src/main/java/de/cxp/ocs/elasticsearch/query/builder/DisMaxQueryFactory.java:31]: This check is not necessary, as the field name is per default null, just remove the if condition.

File: search-service/src/main/java/de/cxp/ocs/elasticsearch/facets/TermFacetCreator.java [hunk 2]
```diff
@@ -30,7 +31,7 @@ public class TermFacetCreator extends NestedFacetCreator {
 	@Setter
 	private int									maxFacetValues	= 100;
 	private final Locale						locale;
-	private final Map<String, FacetEntrySorter>	facetSorters	= new HashMap<>();
+	private final Map<String, FacetEntrySorter>	facetSorters	= new ConcurrentHashMap<>();
```
Comment (gabauer) [inline search-service/src/main/java/de/cxp/ocs/elasticsearch/facets/TermFacetCreator.java:34]: Why must this be now a ConcurrentHashMap?

**Review by gabauer — APPROVED** [review]
LGTM, no critical issues, congrats @nelsonfrz !

File: search-service/src/main/java/de/cxp/ocs/elasticsearch/facets/TermFacetCreator.java (hunk 2 above)
Comment (rudibatt) [reply search-service/src/main/java/de/cxp/ocs/elasticsearch/facets/TermFacetCreator.java]: This is my commit. I changed the PR target to remove this commit from here.
But as an explanation: It fixes a ConcurrentModificationException we experienced at a customer. See [commit message](https://github.com/CommerceExperts/open-commerce-search/commit/4b65b35874fe9ce009980e2612fbd479ebb0d637).

**Review by rudibatt — CHANGES_REQUESTED** [review]

**Pull request merged**